src/runtime_src/core/edge/sw_em/generic_pcie_hal2/halapi.cxx

```diff
@@ -107,7 +112,7 @@ void xclClose(xclDeviceHandle handle)
     return ;
   drv->xclClose();
```

Comment:
This may close the similation/device process where other thread is still using the device. You need to check the usage  of this device .

Reply:
Similar comments for both sw_emu and hw_emu drivers.

Reply:
Based on the review, it seems we require extensive verification before merging the change